File: WinrmCmd.py

#C:\Users\null\Desktop\pywinrm>python test1.py
#win-\k8gege
import winrm
s=winrm.Session('http://192.168.1.116',auth=('k8gege','k8gege520'))#2012 ok
#s=winrm.Session('http://192.168.1.20',auth=('k8gege','k8gege520'))#win7 fail
r=s.run_ps('dir')
r=s.run_cmd('whoami') 
print r.std_out 
print  r.std_err

File: WinrmCmd.go
package main
import (
  "github.com/masterzen/winrm"
  "fmt"
  "os"
  "strconv"
)
//Winrm Remote Shell by k8gege
//http://k8gege.org/Ladon/WinrmScan.html
#C:\Users\k8gege\Desktop\>winrmcmd.exe 192.168.1.116 5985 k8gege k8gege520 whoami
#k8gege

var help = func () {
    fmt.Println("Winrm Shell by k8gege")
    fmt.Println("====================================================")
    fmt.Println("winrmcmd host port user pass cmd")
}

func main() {

	    args := os.Args
    if len(args) < 5 || args == nil {
		help()
        return
    }
	host := args[1]
	port,err := strconv.Atoi(args[2])
	user := args[3]
	pass := args[4]
	cmd := args[5]
	
	endpoint := winrm.NewEndpoint(host, port, false, false, nil, nil, nil, 0)
	client, err := winrm.NewClient(endpoint, user, pass)
	if err != nil {
		panic(err)
	}
client.Run(cmd, os.Stdout, os.Stderr)
}